### PR TITLE
[Tooling] Re-enable localization linting

### PR DESCRIPTION
**Pull request by AliSoftware**
### What

This re-enables the calls to the `ios_lint_localizations` action, which had been disabled in the past due to the fact that one of our keys were too long and thus came back truncated from GlotPress, which limits the keys to 256 characters.

This has since been addressed by our recent improvements in our Localization Tooling (ref: paaHJt-3bh-p2) which means we are now able to properly fix that problematic string (without having to resort to some workaround in the code like splitting the copy in two separate keys) by using a key that is not the English copy 🎉 

### To Test

 - Run `bundle exec fastlane run ios_lint_localizations input_dir:'WordPress/Resources'` to test the action (recently fixed by https://github.com/wordpress-mobile/release-toolkit/pull/353)

At that point given the state of GlotPress, you should get the following violations printed:
```
[23:16:00]: Inconsistencies found between 'en' and 'nl':

`<strong>Once</strong> a week at %@` expected placeholders for [String] but found [] instead.
`<strong>Twice</strong> a week at %@` expected placeholders for [String] but found [] instead.

[23:16:00]: Inconsistencies found between 'en' and 'ro':

`%1$d Like` expected placeholders for [Int] but found [] instead.
`_You and %1$d blogger_ like this.` expected placeholders for [Int] but found [] instead.

[23:16:00]: Inconsistencies found between 'en' and 'zh-Hans':

`To use this app for %@ you'll need to have the Jetpack plugin installed and activated.` expected placeholders for [String] but found [] instead.


[!] Inconsistencies found during Localization linting. Aborting.
```

Those seem to be genuine violations, that we will need to fix at their root in GlotPress first (or ask Polyglots and +i18n to help us do so) before merging that PR (or at least before we do the next beta after merging that PR).

### PR Status

 - This PR points to a branch of the `release-toolkit`. We need to update the reference in the `Pluginfile` once we do a new release of the toolkit, or at least once we merge https://github.com/wordpress-mobile/release-toolkit/pull/353 to trunk and can point to it.
 - We should address the remaining existing violations by fixing the translations in GlotPress first, before merging this PR, otherwise the linter will (rightfully) start complaining about them on every beta. Hence the "DO NOT MERGE" label.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/People/InvitePersonViewController.swift
Comment (AliSoftware) [inline WordPress/Classes/ViewRelated/People/InvitePersonViewController.swift:393]: Thanks to new ability of new tooling to finally support keys-that-are-not-the-English-copy 🎉  paNNhX-nP-p2

File: WordPress/Resources/ar.lproj/Localizable.strings
Comment (AliSoftware) [inline WordPress/Resources/ar.lproj/Localizable.strings]: Fixed all of those manually (search-and-replace) even if those files will be overwritten during next code freeze then after next beta downloading the latest `Localizable.strings` for all locales from GllotPress.

The rationale being that even though those will be overwritten, manually fixing them here in the meantime will allow for a smooth transition between re-enabling the linter and having that key do the round trip to get fixed in GlotPress. We wouldn't want the linter being re-enabled creating blockers during the transition period between this PR being merged and them going to GlotPress and back.

See f3e2a952e26010d17efa2d11d85db3c16c34f276 for full rationale.

File: fastlane/Pluginfile
Comment (AliSoftware) [inline fastlane/Pluginfile]: 🚨  We need to point back to `trunk` or to the next release once https://github.com/wordpress-mobile/release-toolkit/pull/353 lands before merging this PR

File: fastlane/lanes/release.rb
Comment (AliSoftware) [inline fastlane/lanes/release.rb:88]: No more! 🎉

**Comment by AliSoftware** [comment]
Violations found by linter and that will have to be fixed at their root in GlotPress:
 - `nl`: [here](https://translate.wordpress.org/projects/apps/ios/dev/nl/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=12374934&filters%5Btranslation_id%5D=87369788) and [here](https://translate.wordpress.org/projects/apps/ios/dev/nl/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=12374933&filters%5Btranslation_id%5D=87369783) 
 - `ro`: [here](https://href.li/?https://translate.wordpress.org/projects/apps/ios/dev/ro/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=11994998&sort%5Bby%5D=translation_date_added&sort%5Bhow%5D=asc) and [here](https://href.li/?https://translate.wordpress.org/projects/apps/ios/dev/ro/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=12309997&sort%5Bby%5D=translation_date_added&sort%5Bhow%5D=asc)
 - `zh-Hans`: [here](https://translate.wordpress.org/projects/apps/ios/dev/zh-cn/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=11849046&filters%5Btranslation_id%5D=84259945) (seems to be using `U+FF05 FULLWIDTH PERCENT SIGN character: ％` instead of `U+0025 PERCENT SIGN character: %` before the placeholder in its translation?)

File: WordPress/Resources/ar.lproj/Localizable.strings
Comment (mokagio) [reply WordPress/Resources/ar.lproj/Localizable.strings]: >  even though those will be overwritten, manually fixing them here in the meantime will allow for a smooth transition

Always good approach to take 👍

**Review by mokagio — COMMENTED** [review]
Working as expected.

image

Holding off approval till the release toolkit version update to avoid accidental merges (just as an extra measure, despite the "Do Not Merge" label being already in place)

**Comment by mokagio** [comment]
I had a look at the violations in GlotPress to see if I could address them. The `ro` and `nl` had genuinely missing placeholders. I fixed the `ro` ones and asked for help for the `nl` ones (pxLjZ-6Z4-p2). But the `zh-Hans` (`zh-cn` in GlotPress) surprises me:

Our error:

```
[13:28:57]: Inconsistencies found between 'en' and 'zh-Hans':

`To use this app for %@ you'll need to have the Jetpack plugin installed and activated.` expected placeholders for [String] but found [] instead.
```

How the translation is [on GlotPress](https://translate.wordpress.org/projects/apps/ios/dev/zh-cn/default/?filters%5Bterm%5D=need+to+have+the+Jetpack&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc):

image

Could the violation be due to the `,` after the `%@` in the translation? As far as [I can see](https://translate.wordpress.org/projects/apps/ios/dev/zh-cn/default/?filters%5Bterm%5D=%25%40&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc&page=1) this translation is the only one that uses `%@,`. 

I wonder, then, if this is a false positive. 🤔

**Comment by AliSoftware** [comment]
@mokagio Thanks for looking into those and opening the +i18n post to start getting those fixed 💙

> I wonder, then, if this is a false positive. 🤔

As I mentioned in my comment above, this one is a genuine violation too… albeit a more subtle one to see (which is another reason why the linter is so useful, being able to catch it while the human eye would easily miss it):

>  - `zh-Hans`: [here](https://translate.wordpress.org/projects/apps/ios/dev/zh-cn/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=11849046&filters%5Btranslation_id%5D=84259945) (seems to be using `U+FF05 FULLWIDTH PERCENT SIGN character: ％` instead of `U+0025 PERCENT SIGN character: %` before the placeholder in its translation?)

**Comment by AliSoftware** [comment]
> I fixed the `ro` ones

Your suggestions apparently got rejected by a contributor in GlotPress since 😞

**Comment by mokagio** [comment]
> As I mentioned in my comment above, this one is a genuine violation too… albeit a more subtle one to see (which is another reason why the linter is so useful, being able to catch it while the human eye would easily miss it):

Oh, I had missed that comment in the stream of other ones in the PR. Sorry. 👍

**Comment by mokagio** [comment]
I'll be code freezing 19.7 in ~8 hours, but I'm going to leave this in the 19.7 milestone even though it's still waiting for a new version of the release-toolkit to ship. 

This change affects code that runs when shipping betas or finalizing the build, so it makes sense to keep it for this release updating the base branch and test it live sooner.

**Comment by AliSoftware** [comment]
> This change affects code that runs when shipping betas or finalizing the build, so it makes sense to keep it for this release updating the base branch and test it live sooner.

@mokagio I'll let you rebase this PR on `release/19.7` on your day (didn't have time to do it myself during my day).

---

One thing to note btw, is that one part of the fix from this PR is to us a different key for the >256 characters copy that previously made the linter fail. So if you rebase this on top of `release/19.7` _after_ having done the code freeze, that means the new key will only start replacing the old one only next time you merge the `release/*` branch to `trunk`… by which time the strings would likely already have been sent to our translation vendor via the WeeklyKit for the Mag16 translations, and the new key would not make the cut.

That might not be a big deal though, since this is only a change of key, with the copy staying the same, so thanks to GlotPress' "translation memory" feature, GlotPress will automatically detect the similarities of the copies between old and new key, and those will be only a click on a button away from being translated, making it still likely for those to land it time even for Mag16 (as the community will likely be the ones clicking on that button). Worse case scenario we'd use our superpowers (i18n account) to approve the translations for all Mag16 for that new key (provided those translations would be the same as the ones used for the old key, and polyglots used "translation memory" instead of re-translating them from scratch…)

**Comment by AliSoftware** [comment]
@mokagio I've rebased the PR on top of `release/19.7` again to fix the conflict that started to appear after you did a new beta yesterday, and [dropped my manual update of the `.strings` in this PR accordingly](https://github.com/wordpress-mobile/WordPress-iOS/pull/18348#discussion_r847910097)(†).

This is thus now ready to for review, and to be merged whenever you're ready.

---

(†) The creation of the new beta updated the `Localizable.strings` with latest translations, as expected… which created a conflict with the manual temporary fix I did back in [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/18348#discussion_r847910097).

Since that this PR is now targeting `release/19.7` and is thus way after the code-freeze happened, I've decided to drop those commits (the ones which did a find/replace of the `Localizable.strings` with the new <256 key). The new, old key will just continue to be used during this 19.7 sprint, and the shorter key will just be picked up during next code freeze.

#### Rationale:

Given the timing and change of target branch that happened since, I think this is better than trying to replace the key now:
 - Because such a change would be overwritten again if/when you do yet another beta before the end of sprint (well… or when you'll do the final release)
 - And if we replaced the key in the `en.lproj`, once then imported in GlotPress, that would make the newly imported key not yet have any translation, "losing" the ones we currently have for that copy, and thus creating a regression for other locales for that copy in the app
 - And while GlotPress' "translation memory" would make it easy for polyglots to reuse the translations from the old key, there's no guarantee that this will actually happen before end of sprint given we're now at End of Week 1

**Comment by AliSoftware** [comment]
PS: Note that given I didn't fix (find/replace) the `*.lproj/Localizable.strings` files with the new key anymore — and they are thus still using the old key, the execution of `fastlane run ios_lint_localizations input_dir:'WordPress/Resources'` (which is run during `new_beta_release` and `finalize_release`)… will currently fail.

I see multiple options to circumvent that during the transition:
 - Keep the PR as is, and when you'll get the errors next time you do a beta or the final release during this sprint, manually fix the `*.strings` files once they have been downloaded before continuing
    - The `allow_retry:true` option we've enabled on the `ios_lint_localization` call will allow you to do that without issue nor disrupting your beta / final-release flows, that's a use case the linter was built to handle after all 🙂.
    - All you'd need to do to fix that during the retry chance that the linter will give you would be a simple find/replace `"the long key" =` with `"the short key" = ` in the `.strings` files of all `*.lproj`s.
 - Update the key in the `en.lproj/Localizable.strings` in this PR, and soon after it's merged — i.e once the new key has been imported in GlotPress — manually go use the "Translation Memory" feature of GlotPress for that key, telling it to reuse the copy from the old key as the one for the new one.
   - One big advantage of doing that ourselves is that we guarantee that the change of key won't make the new key miss translations in some locales (the ones where polyglots are less active) for a while until they pick up the pace, and also guarantee that the copy won't be retranslated from scratch by polyglots who might not realize that there would already be a suggesting in the Translation Memory one click away from being reused.
   - That would be a bit cumbersome to have to manually do this for each of the locales we have in the GlotPress project for that key. But while a bit systematic, that would be very straightforward, so I think that'd be worth it
 - Split this PR in two, one which just updates the key to the shorter one but keeps the linter disabled until next code-freeze once the new key is sent to GlotPress.
    - That would require us to once-again rebase/rework the commits from that PR, and also postpone landing those changes to 19.9 (change key in 19.8, enable linter in 19.9)…
    - …and I'm not a fan of keeping that open for too long, especially since we want to wrap up the Localization Project by this sprint anyway

---

My preference would probably be option 2. It would be a bit more cumbersome (except if Team Global has some magic up their sleeve to automate it?) but slightly better and more sustainable. Option 1 might look easier to put in practice… but would have to be redone on every beta/final build of this sprint until we reach next code freeze… at which point we'd probably want to do Option 2 anyway…

I'll let you weigh those options and decide accordingly. After all, as Release Manager of WPiOS you're the one who will have to deal with those anyway 😅

**Comment by mokagio** [comment]
I choose... Option 2.

Better to deal with the complexity ASAP that add extra work to the unpredictable number of beta builds we'll do for this version. And while in the worse case of one build per day that would be "only" six builds, I can't guarantee I'll be around to do them or that there won't be any time pressure. So sorting this out ASAP seems a more future proof approach.

**Comment by mokagio** [comment]
> I choose... Option 2.

FYI. I haven't finished this yet because I went down several rabbit holes while looking at the other lint issues, one of those even resulted in a PR: #18424.

I need to take a long break now. I'll hopefully get to this later today.

**Review by mokagio — APPROVED** [review]
As [discussed](https://github.com/wordpress-mobile/WordPress-iOS/pull/18348#issuecomment-1105945769), the key value has been update in the `en.lproj/Localizable.strings` file  via [`8ac6584`](https://github.com/wordpress-mobile/WordPress-iOS/pull/18348/commits/8ac6584469211cf5206b853b752c5afc119b3f84).

There other linter errors already mentioned in the PR are still there, and I'm working with the polyglots to address them.

```
[20:32:47]: Inconsistencies found between 'en' and 'nl':

`<strong>Once</strong> a week at %@` expected placeholders for [String] but found [] instead.
`<strong>Twice</strong> a week at %@` expected placeholders for [String] but found [] instead.

[20:32:47]: Inconsistencies found between 'en' and 'ro':

`%1$d Like` expected placeholders for [Int] but found [] instead.
`_You and %1$d blogger_ like this.` expected placeholders for [Int] but found [] instead.

[20:32:47]: Inconsistencies found between 'en' and 'zh-Hans':

`To use this app for %@ you'll need to have the Jetpack plugin installed and activated.` expected placeholders for [String] but found [] instead.
```

---

Approving. Auto-merge will do the rest.

**Pull request merged**
